mixer/pkg/config/crd/store.go

```diff
@@ -204,17 +204,19 @@ func (s *Store) retryCreateCache(
 	tick := time.Tick(s.bgRetryInterval)
 	stopRetry := false
 
-	for len(remaining) != 0 && !stopRetry {
+	for range tick {
 		select {
 		case <-s.donec:
 			stopRetry = true
-		case <-tick:
+		default:
 			rm := s.checkAndCreateCaches(d, lwBuilder, remaining)
 			if len(rm) < len(remaining) {
 				log.Debugf("discovered %v new kinds, remaining undiscovered kinds: %v", len(remaining)-len(rm), rm)
 			}
 			remaining = rm
-		default:
```

Comment:
Simply having an empty `default` will potentially cause a tight loop, so removing `default` case will also help and will be minimal change. 
how did you test this?

Reply:
Why does removing `default` help? All channel cases will still be checked without any interval between loops right? By changing loop condition into `range tick`, number of channel reads reduces, so does cpu.

I repro the cpu issue and fixed it with this change. I cannot find a good way to add an unit test for this change tho. The unit test for the origin change was to test if retry interval is respected by checking if the total execution time of background retry longer than a certain period.

Reply:
Agree, remove default can also solve it.

Reply:
Peter, selecting on multiple blocking channels will not cause a tight loop. Select blocks until one of the channel fires. When default is added, default always fires and causes the tight loop.

Reply:
I see. Thanks!